spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -42,13 +44,25 @@
 
 	private final PulsarProducerFactory<T> producerFactory;
 
+	private final List<ProducerInterceptor> interceptors;
+
 	/**
 	 * Constructs a template instance.
 	 * @param producerFactory the producer factory used to create the backing Pulsar
 	 * producers.
 	 */
 	public PulsarTemplate(PulsarProducerFactory<T> producerFactory) {
+		this(producerFactory, null);
+	}
+
+	/**
+	 * Constructs a template instance.
+	 * @param producerFactory the producer factory used to create the backing Pulsar producers.
+	 * @param interceptors the {@link ProducerInterceptor}s to add to the producer.
+	 */
+	public PulsarTemplate(PulsarProducerFactory<T> producerFactory, List<ProducerInterceptor> interceptors) {
```

Comment:
> Rather than taking a collection of ProducerInterceptor on PulsarTemplate, how about providing a setter for a single interceptor? I think that is probably the majority of the use case. For multiple interceptors on the same template, how about introducing another indirection - something like a CompositeProducerInterceptor that implements the ProducerInterceptor? In this CompositeProduceInterceptor, we can set the delegate interceptors in the order the user wants and then call each interceptor.

@sobychacko I do understand keeping it simple and not adding things we may not need. In this case, because of the natural fit w/ "N things" that the [`ObjectProvider`](https://github.com/spring-projects-experimental/spring-pulsar/pull/59#discussion_r942860574) provides as well as the fact that the underlying Pulsar API accept "N things" I would be in favor of keeping "N" from the start.

Reply:
Ok, that makes sense. `ObjectProvider` usgae is nice.